File: android/app/src/main/java/com/lichso/app/ui/screen/splash/SplashScreen.kt
package com.lichso.app.ui.screen.splash

import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.draw.scale
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.SpanStyle
import androidx.compose.ui.text.buildAnnotatedString
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.lichso.app.BuildConfig
import com.lichso.app.R
import kotlinx.coroutines.delay

// ══════════════════════════════════════════
// Palette — Traditional Red & Gold
// ══════════════════════════════════════════
private val BgDeepRed = Color(0xFF6D0B0B)
private val BgRed = Color(0xFF9B1B1B)
private val BgCrimson = Color(0xFFC62828)

private val Gold50 = Color(0xFFFFF8E1)   // near-white gold highlight
private val Gold200 = Color(0xFFE6C361)
private val Gold400 = Color(0xFFD4A017)
private val Gold600 = Color(0xFFB8860B)
private val Gold800 = Color(0xFF8B6914)

private val ZenixStart = Color(0xFF64FFDA)
private val ZenixMid = Color(0xFF448AFF)
private val ZenixEnd = Color(0xFFB388FF)

@Composable
fun SplashScreen(
    onSplashFinished: () -> Unit
) {
    /* ─── animation flags ─── */
    var started by remember { mutableStateOf(false) }

    // 1) Logo — spring bounce
    val logoScale by animateFloatAsState(
        targetValue = if (started) 1f else 0f,
        animationSpec = spring(dampingRatio = .5f, stiffness = 280f),
        label = "logoScale"
    )
    val logoAlpha by animateFloatAsState(
        targetValue = if (started) 1f else 0f,
        animationSpec = tween(500), label = "logoAlpha"
    )

    // 2) Title — delayed fade
    val titleAlpha by animateFloatAsState(
        targetValue = if (started) 1f else 0f,
        animationSpec = tween(500, delayMillis = 350), label = "titleAlpha"
    )
    val titleSlide by animateFloatAsState(
        targetValue = if (started) 0f else 24f,
        animationSpec = tween(500, delayMillis = 350, easing = FastOutSlowInEasing),
        label = "titleSlide"
    )

    // 3) Subtitle — later
    val subAlpha by animateFloatAsState(
        targetValue = if (started) 1f else 0f,
        animationSpec = tween(400, delayMillis = 650), label = "subAlpha"
    )
    val subSlide by animateFloatAsState(
        targetValue = if (started) 0f else 20f,
        animationSpec = tween(400, delayMillis = 650, easing = FastOutSlowInEasing),
        label = "subSlide"
    )

    // 4) Bottom — latest
    val bottomAlpha by animateFloatAsState(
        targetValue = if (started) 1f else 0f,
        animationSpec = tween(500, delayMillis = 900), label = "bottomAlpha"
    )

    // ∞ Slow-spinning gold ring
    val inf = rememberInfiniteTransition(label = "inf")
    val ringRotation by inf.animateFloat(
        initialValue = 0f, targetValue = 360f,
        animationSpec = infiniteRepeatable(tween(20_000, easing = LinearEasing)),
        label = "ringRot"
    )
    val glowPulse by inf.animateFloat(
        initialValue = .06f, targetValue = .14f,
        animationSpec = infiniteRepeatable(
            tween(3000, easing = EaseInOutSine), RepeatMode.Reverse
        ), label = "glow"
    )

    LaunchedEffect(Unit) {
        started = true
        delay(2400)
        onSplashFinished()
    }

    // ══════════════════════════════════════════
    //  UI
    // ══════════════════════════════════════════
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(
                Brush.verticalGradient(
                    colors = listOf(BgDeepRed, BgRed, BgCrimson),
                )
            ),
        contentAlignment = Alignment.Center
    ) {
        /* ── ambient glow top-right ── */
        Box(
            Modifier
                .size(340.dp)
                .align(Alignment.TopEnd)
                .offset(x = 100.dp, y = (-80).dp)
                .alpha(glowPulse)
                .background(
                    Brush.radialGradient(
                        colors = listOf(Gold400, Color.Transparent),
                        radius = 340f
                    ),
                    CircleShape
                )
        )
        /* ── ambient glow bottom-left ── */
        Box(
            Modifier
                .size(280.dp)
                .align(Alignment.BottomStart)
                .offset(x = (-70).dp, y = 100.dp)
                .alpha(glowPulse * .6f)
                .background(
                    Brush.radialGradient(
                        colors = listOf(Gold200, Color.Transparent),
                        radius = 280f
                    ),
                    CircleShape
                )
        )

        /* ═══ center content ═══ */
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier.offset(y = (-28).dp)
        ) {
            /* ── logo + ring ── */
            Box(
                contentAlignment = Alignment.Center,
                modifier = Modifier.size(210.dp)
            ) {
                // Rotating thin gold arc ring
                Canvas(
                    modifier = Modifier
                        .matchParentSize()
                        .rotate(ringRotation)
                        .alpha(.35f)
                ) {
                    val stroke = Stroke(width = 2.dp.toPx(), cap = StrokeCap.Round)
                    drawArc(
                        brush = Brush.sweepGradient(
                            listOf(Color.Transparent, Gold200, Gold50, Color.Transparent)
                        ),
                        startAngle = 0f,
                        sweepAngle = 270f,
                        useCenter = false,
                        style = stroke
                    )
                }

                // Soft radial glow behind logo
                Box(
                    Modifier
                        .size(180.dp)
                        .alpha(.12f)
                        .background(
                            Brush.radialGradient(listOf(Gold200, Color.Transparent)),
                            CircleShape
                        )
                )

                // Logo
                Image(
                    painter = painterResource(R.drawable.splash_logo),
                    contentDescription = "Lịch Số",
                    contentScale = ContentScale.Fit,
                    modifier = Modifier
                        .size(160.dp)
                        .scale(logoScale)
                        .alpha(logoAlpha)
                )
            }

            /* ── app name ── */
            Text(
                text = "Lịch Số",
                style = TextStyle(
                    fontSize = 28.sp,
                    fontWeight = FontWeight.Bold,
                    brush = Brush.linearGradient(
                        listOf(Gold600, Gold200, Gold50, Gold200, Gold600),
                        start = Offset(0f, 0f),
                        end = Offset(Float.POSITIVE_INFINITY, 0f)
                    ),
                    letterSpacing = 4.sp,
                    textAlign = TextAlign.Center,
                ),
                modifier = Modifier
                    .graphicsLayer {
                        alpha = titleAlpha
                        translationY = titleSlide
                    }
            )

            Spacer(Modifier.height(10.dp))

            /* ── thin divider ── */
            Box(
                Modifier
                    .width(56.dp)
                    .height(2.dp)
                    .alpha(titleAlpha)
                    .background(
                        Brush.horizontalGradient(
                            listOf(Color.Transparent, Gold400, Color.Transparent)
                        ),
                        RoundedCornerShape(1.dp)
                    )
            )

            Spacer(Modifier.height(14.dp))

            /* ── tagline ── */
            Text(
                text = "Lịch Vạn Niên  ·  Gia Phả  ·  AI Tử Vi",
                style = TextStyle(
                    fontSize = 13.sp,
                    fontWeight = FontWeight.Normal,
                    color = Color.White.copy(alpha = .55f),
                    letterSpacing = .8.sp,
                    textAlign = TextAlign.Center,
                ),
                modifier = Modifier.graphicsLayer {
                    alpha = subAlpha
                    translationY = subSlide
                }
            )

            Spacer(Modifier.height(4.dp))

            Text(
                text = "v${BuildConfig.VERSION_NAME}",
                style = TextStyle(
                    fontSize = 11.sp,
                    color = Color.White.copy(alpha = .25f),
                ),
                modifier = Modifier.graphicsLayer {
                    alpha = subAlpha
                    translationY = subSlide
                }
            )
        }

        /* ═══ bottom branding ═══ */
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier
                .align(Alignment.BottomCenter)
                .padding(bottom = 52.dp)
                .alpha(bottomAlpha)
        ) {
            // Small decorative dot
            Box(
                Modifier
                    .size(3.dp)
                    .background(Gold400.copy(alpha = .4f), CircleShape)
            )
            Spacer(Modifier.height(10.dp))
            Text(
                text = buildAnnotatedString {
                    withStyle(
                        SpanStyle(
                            brush = Brush.linearGradient(
                                listOf(Gold600, Gold200, Gold50, Gold200, Gold600),
                                start = Offset(0f, 0f),
                                end = Offset(Float.POSITIVE_INFINITY, 0f)
                            ),
                        )
                    ) { append("Phát triển bởi  ") }
                    withStyle(
                        SpanStyle(
                            brush = Brush.linearGradient(
                                listOf(ZenixStart, ZenixMid, ZenixEnd),
                                start = Offset(0f, 0f),
                                end = Offset(Float.POSITIVE_INFINITY, 0f)
                            ),
                            fontWeight = FontWeight.SemiBold,
                        )
                    ) { append("Zenix Labs") }
                },
                style = TextStyle(
                    fontSize = 12.sp,
                    letterSpacing = .5.sp,
                ),
            )
        }
    }
}

File: ios/LichSo/LichSo/Screen/Splash/SplashScreen.swift
import SwiftUI

// ═══════════════════════════════════════════
// Splash Screen — Port from Android
// Traditional Red & Gold theme
// ═══════════════════════════════════════════

private let BgDeepRed = Color(hex: "6D0B0B")
private let BgRed = Color(hex: "9B1B1B")
private var BgCrimson: Color { LSTheme.badRed }

private let Gold50 = Color(hex: "FFF8E1")
private let Gold200 = Color(hex: "E6C361")
private var Gold400: Color { LSTheme.gold }
private let Gold600 = Color(hex: "B8860B")
private let Gold800 = Color(hex: "8B6914")

private let ZenixStart = Color(hex: "64FFDA")
private let ZenixMid = Color(hex: "448AFF")
private let ZenixEnd = Color(hex: "B388FF")

struct SplashScreen: View {
    let onSplashFinished: () -> Void

    @State private var started = false
    @State private var logoScale: CGFloat = 0
    @State private var logoOpacity: Double = 0
    @State private var titleOpacity: Double = 0
    @State private var titleOffset: CGFloat = 24
    @State private var subOpacity: Double = 0
    @State private var subOffset: CGFloat = 20
    @State private var bottomOpacity: Double = 0
    @State private var ringRotation: Double = 0
    @State private var glowPulse: Double = 0.06

    var body: some View {
        ZStack {
            // ── Background gradient ──
            LinearGradient(
                colors: [BgDeepRed, BgRed, BgCrimson],
                startPoint: .top,
                endPoint: .bottom
            )
            .ignoresSafeArea()

            // ── Ambient glow top-right ──
            Circle()
                .fill(
                    RadialGradient(
                        colors: [Gold400, Color.clear],
                        center: .center,
                        startRadius: 0,
                        endRadius: 170
                    )
                )
                .frame(width: 340, height: 340)
                .opacity(glowPulse)
                .offset(x: 100, y: -340)

            // ── Ambient glow bottom-left ──
            Circle()
                .fill(
                    RadialGradient(
                        colors: [Gold200, Color.clear],
                        center: .center,
                        startRadius: 0,
                        endRadius: 140
                    )
                )
                .frame(width: 280, height: 280)
                .opacity(glowPulse * 0.6)
                .offset(x: -70, y: 340)

            // ═══ Center content ═══
            VStack(spacing: 0) {
                // ── Logo + ring ──
                ZStack {
                    // Rotating gold arc ring
                    Circle()
                        .trim(from: 0, to: 0.75)
                        .stroke(
                            AngularGradient(
                                colors: [Color.clear, Gold200, Gold50, Color.clear],
                                center: .center
                            ),
                            lineWidth: 2
                        )
                        .frame(width: 210, height: 210)
                        .rotationEffect(.degrees(ringRotation))
                        .opacity(0.35)

                    // Soft glow behind logo
                    Circle()
                        .fill(
                            RadialGradient(
                                colors: [Gold200, Color.clear],
                                center: .center,
                                startRadius: 0,
                                endRadius: 90
                            )
                        )
                        .frame(width: 180, height: 180)
                        .opacity(0.12)

                    // App logo image
                    Image("AppLogo")
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                        .frame(width: 140, height: 140)
                        .shadow(color: Color.black.opacity(0.35), radius: 24, x: 0, y: 12)
                        .scaleEffect(logoScale)
                        .opacity(logoOpacity)
                }
                .frame(width: 210, height: 210)

                // ── App name ──
                Text("Lịch Số")
                    .font(.system(size: 28, weight: .bold, design: .serif))
                    .foregroundStyle(
                        LinearGradient(
                            colors: [Gold600, Gold200, Gold50, Gold200, Gold600],
                            startPoint: .leading,
                            endPoint: .trailing
                        )
                    )
                    .tracking(4)
                    .opacity(titleOpacity)
                    .offset(y: titleOffset)

                Spacer().frame(height: 10)

                // ── Thin divider ──
                Rectangle()
                    .fill(
                        LinearGradient(
                            colors: [Color.clear, Gold400, Color.clear],
                            startPoint: .leading,
                            endPoint: .trailing
                        )
                    )
                    .frame(width: 56, height: 2)
                    .opacity(titleOpacity)

                Spacer().frame(height: 14)

                // ── Tagline ──
                Text("Lịch Vạn Niên  ·  Gia Phả  ·  AI Tử Vi")
                    .font(.system(size: 13, weight: .regular))
                    .foregroundColor(.white.opacity(0.55))
                    .tracking(0.8)
                    .opacity(subOpacity)
                    .offset(y: subOffset)

                Spacer().frame(height: 4)

                Text("v1.0.0")
                    .font(.system(size: 11))
                    .foregroundColor(.white.opacity(0.25))
                    .opacity(subOpacity)
                    .offset(y: subOffset)
            }
            .offset(y: -28)

            // ═══ Bottom branding ═══
            VStack(spacing: 10) {
                Spacer()

                Circle()
                    .fill(Gold400.opacity(0.4))
                    .frame(width: 3, height: 3)

                HStack(spacing: 4) {
                    Text("Phát triển bởi  ")
                        .font(.system(size: 12))
                        .foregroundStyle(
                            LinearGradient(
                                colors: [Gold600, Gold200, Gold50, Gold200, Gold600],
                                startPoint: .leading,
                                endPoint: .trailing
                            )
                        )
                        .tracking(0.5)

                    Text("Zenix Labs")
                        .font(.system(size: 12, weight: .semibold))
                        .foregroundStyle(
                            LinearGradient(
                                colors: [ZenixStart, ZenixMid, ZenixEnd],
                                startPoint: .leading,
                                endPoint: .trailing
                            )
                        )
                }
                .padding(.bottom, 52)
            }
            .opacity(bottomOpacity)
        }
        .onAppear {
            startAnimations()
        }
    }

    private func startAnimations() {
        // Logo spring bounce
        withAnimation(.spring(response: 0.5, dampingFraction: 0.5, blendDuration: 0)) {
            logoScale = 1.0
            logoOpacity = 1.0
        }

        // Title fade in (delayed 350ms)
        withAnimation(.easeOut(duration: 0.5).delay(0.35)) {
            titleOpacity = 1.0
            titleOffset = 0
        }

        // Subtitle (delayed 650ms)
        withAnimation(.easeOut(duration: 0.4).delay(0.65)) {
            subOpacity = 1.0
            subOffset = 0
        }

        // Bottom (delayed 900ms)
        withAnimation(.easeOut(duration: 0.5).delay(0.9)) {
            bottomOpacity = 1.0
        }

        // Infinite ring rotation
        withAnimation(.linear(duration: 20).repeatForever(autoreverses: false)) {
            ringRotation = 360
        }

        // Glow pulse
        withAnimation(.easeInOut(duration: 3).repeatForever(autoreverses: true)) {
            glowPulse = 0.14
        }

        // Navigate after 2.4s
        DispatchQueue.main.asyncAfter(deadline: .now() + 2.4) {
            onSplashFinished()
        }
    }
}

#Preview {
    SplashScreen(onSplashFinished: {})
}
